### DISCUSSION TITLE
The magnetic field looks a bit strange

### DISCUSSION THREAD

**Discussion by sm-liang** [PrincetonUniversity/athena#551] [Q&A] [answered]
I want apply the Bx = b0* tanh(y/delta), the harris-current configure, and there is the magnetic field code of ProblemGenerator
if (MAGNETIC_FIELDS_ENABLED) {
    for (int k=ks; k<=ke; ++k) {
      for (int j=js; j<=je; ++j) {
        for (int i=is; i<=ie+1; ++i) {
          y0 = pcoord->x2v(j);
          pfield->b.x1f(k,j,i) = b0 * tanh(y0/width);
          if (i>is){
            bx0 = 0.5*(pfield->b.x1f(k,j,i)+pfield->b.x1f(k,j,i-1));
            phydro->u(IEN,k,j,i-1) += (bx0*bx0)/2.0;
          }
        }
      }
    }
    for (int k=ks; k<=ke; ++k) {
      for (int j=js; j<=je+1; ++j) {
        for (int i=is; i<=ie; ++i) {
          pfield->b.x2f(k,j,i) = 0.0;
        }
      }
    }
    for (int k=ks; k<=ke+1; ++k) {
      for (int j=js; j<=je; ++j) {
        for (int i=is;i<=ie;++i) {
          pfield->b.x3f(k,j,i) = 0.0;
        }
      }
    }
  }
where b0=1, but when I imshow the initial magnetic field, it looks like 
![B-field](https://github.com/PrincetonUniversity/athena/assets/82441061/9bcb1226-5c1f-43b1-91ea-aab9fb745127)
there are slices of Bx, there are Bx[mid,:,:], Bx[:,mid,:], Bx[:,:,mid].
And I check the By and Bz, they are ok, both them are 0.
Please tell me where the problem is. 
The attach file is my input file.
[input.txt](https://github.com/PrincetonUniversity/athena/files/13761576/input.txt)

**Comment by sm-liang** [comment]
This is the ProblemGenerator file.
[turb_mhd_iso.txt](https://github.com/PrincetonUniversity/athena/files/13761639/turb_mhd_iso.txt)

**Comment by c-white** [comment, marked as answer]
If you are compiling with `--eos isothermal` like your input file claims, then your problem generator has a bug. It's modifying `phydro->u(IEN,...)` when this variable does not exist, so it's overwriting some other part of memory. This part needs to be protected with `if (NON_BAROTROPIC_EOS)`.

**Reply by sm-liang** [reply]
It works, Thanks, Chris.
